database/migrations/2023_01_18_201909_create_voting_tables.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class () extends Migration {
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('sittings', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->string('title');
+            $table->datetime('opened_at');
+            $table->datetime('closed_at')->nullable();
+        });
+        Schema::create('questions', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->foreignId('sitting_id')
+                ->references('id')->on('sittings')
+                ->onDelete('cascade');
+            $table->string('title');
+            $table->integer('max_options')->default(1);
+            $table->datetime('opened_at')->nullable();
+            $table->datetime('closed_at')->nullable();
+        });
+        Schema::create('question_options', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->foreignId('question_id')
+                ->references('id')->on('questions')
+                ->onDelete('cascade');
+            $table->string('title');
+            $table->integer('votes')->default(0);
+        });
+        Schema::create('question_user', function (Blueprint $table) {
+            $table->foreignId('question_id')
+                ->references('id')->on('questions')
+                ->onDelete('cascade');
+            $table->foreignId('user_id')
+                ->references('id')->on('users')
+                ->onDelete('cascade');
+            $table->timestamps();
```

Comment:
Not sure this works, but please try adding composite keys.
`$table->index(['question_id', 'user_id']);`
This should enforce in db level that the user can not vote for the same question twice. 
You can try this with inserting rows directly in mysql.

Reply:
Instead of `foreignId('user_id')` you should use `foreignIdFor(User::class)`. The result is probably the same, but it is less likely to make a mistake if you use it (and it is also easier to read).

Reply:
@kdmnk here the problem is that now they do vote multiple times if there is a multiple-choice question. But maybe this could be solved if the vote() function were moved to the Question model, as you have suggested. (Probably you were right with that:D)

Reply:
Hm... but then it really makes sense to move voting to the Question model; it would create a single question-user pair and then increment the votes. I think I will follow that route.